main.go

```diff
@@ -1,7 +1,328 @@
+	var certificate Certificate
+	_ = json.NewDecoder(r.Body).Decode(&certificate)
+	certificate.ID = strconv.Itoa(rand.Intn(1000000)) //Mock ID - not safe
+	certificates = append(certificates, certificate)
+	err := json.NewEncoder(w).Encode(certificate)
+	if err != nil {
+		fmt.Fprintf(w, err.Error())
+	}
+}
+
+// Route handler PATCH certificate/{id} method
+func patchCertificate(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	//Todo (Farouk): PATCH method can update individual fields in certificate
+}
+
+// Route handler PUT certificate/{id} method
+func putCertificate(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range certificates {
+		if item.ID == params["id"] {
+			certificates = append(certificates[:index], certificates[index+1:]...)
+			var certificate Certificate
+			//TODO (Farouk): PUT method: implement error if missing fields
+			_ = json.NewDecoder(r.Body).Decode(&certificate)
+			certificate.ID = params["id"]
+			certificates = append(certificates, certificate)
+			err := json.NewEncoder(w).Encode(certificate)
+			if err != nil {
+				fmt.Fprintf(w, err.Error())
+			}
+			return
+		}
+	}
+}
+
+// Route Handler DELETE certificate/{id} method
+func deleteCertificate(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range certificates {
+		if item.ID == params["id"] {
+			certificates = append(certificates[:index], certificates[index+1:]...)
+			break
+		}
+	}
+	err := json.NewEncoder(w).Encode(certificates)
+	if err != nil {
+		fmt.Fprintf(w, err.Error())
+	}
+}
+
+func createUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var user User
+	_ = json.NewDecoder(r.Body).Decode(&user)
+	user.ID = strconv.Itoa(rand.Intn(1000000)) //Mock ID - not safe
+	users = append(users, user)
+	//jwt?
+	//json.NewEncoder(w).Encode("")
+}
+
+func createTransfer(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range certificates {
+		if item.ID == params["id"] && item.OwnerID == params["userId"] {
+			var certificate Certificate
+			//certificates = append(certificates[:index], certificates[index+1:]...)
+			err := json.NewDecoder(r.Body).Decode(&certificate)
+			if err != nil {
+				fmt.Fprintf(w, err.Error())
+				return
+			}
+			item.Transfer.To = certificate.Transfer.To
+			item.Transfer.Status = "Pending"
+		}
+	}
+}
+
+func acceptTransfer(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range certificates {
+		if item.ID == params["id"] {
+			err := json.NewEncoder(w).Encode(item)
+			if err != nil {
+				fmt.Fprintf(w, err.Error())
+				return
+			}
+			item.OwnerID = params["userId"]
```

Comment:
Same as the `createTransfer` we will need to wrap this in the jwtMiddleWare and ensure we get the userId from the token.